### DISCUSSION TITLE
Kimai 2 & cPanel

### DISCUSSION THREAD

**Discussion by katalux** [kimai/kimai#2761] [On-Premise: Installation & Updates]
Hi Kevin, it is possible to install Kimai with cPanel? There is a problem with cPanel permission on "CREATE DATABASE", the cPanel user hasn't permission of creating db. It's possible to get access with an already created db without the creation process during the installation? Thanks!

**Comment by kevinpapst** [comment]
Why not trying first? 😄

**Reply by katalux** [reply]
Just done with this result

`
Kimai installation running ...`
`[ERROR] Failed to create database: An exception occurred in driver: SQLSTATE[HY000] [1045] Access denied for user "xxxxxxxxx"@"localhost" (using password: YES)
`

**Reply by kevinpapst** [reply]
Hm, I almost 100% certain that it works. 
See the relevant code https://github.com/kevinpapst/kimai2/blob/master/src/Command/InstallCommand.php#L149
It checks if connection can be established: if so, the create step is skipped. 
If connection cannot be established, it still uses the error friendly "only create if not exists" flag.

**Reply by katalux** [reply]
Maybe some errors here?

`DATABASE_URL=mysql://timesheet_user:<password>@127.0.0.1:3306/timesheet_db`

**Reply by kevinpapst** [reply]
Maybe, yes. 
I don't cPanel it, but I know that there are cPanel users that installed Kimai this way.

**Reply by katalux** [reply]
Where do You think exactly is the error?
Password, user and db names are correct.

**Reply by kevinpapst** [reply]
Did you check the log file in Kimai directory var/log/prod.log ?
Likely an permission issue. Can you grant all permissions to the kimai user?

**Reply by katalux** [reply]
There no file in that directory (var/log)
The user timesheet_user as all permissions with the database timesheet_db

**Reply by kevinpapst** [reply]
> Access denied for user "xxxxxxxxx"@"localhost" 

says different

**Reply by katalux** [reply]
Deleted user and re-created it. Now it works! 
Maybe some problems with cPanel user privileges settings :\
